assets/js/blocks/product-collection/inspector-controls/taxonomy-controls/taxonomy-item.tsx

```diff
@@ -0,0 +1,269 @@
+/**
+ * External dependencies
+ */
+import { useEntityRecords } from '@wordpress/core-data';
+import { Taxonomy } from '@wordpress/core-data/src/entity-types';
+import { useState, useMemo, useRef } from '@wordpress/element';
+import { useDebounce } from '@wordpress/compose';
+import { FormTokenField } from '@wordpress/components';
+
+type Term = {
+	id: number;
+	name: string;
+	slug: string;
+};
+
+interface TaxonomyItemProps {
+	taxonomy: Taxonomy;
+	termIds: number[];
+	onChange: ( termIds: number[] ) => void;
+}
+
+// A constant empty array that is reused throughout the component.
+const EMPTY_ARRAY: [] = [];
+
+// Base arguments for querying terms.
+const BASE_QUERY_ARGS = {
+	order: 'asc',
+	_fields: 'id,name,slug',
+	context: 'view',
+};
+
+// Function to get the term id based on user input in the `FormTokenField`.
+const getTermIdByTermValue = (
+	searchTerm: Term | string,
+	termNameToIdMap: Map< string, number >
+): number | undefined => {
+	const termId = ( searchTerm as Term )?.id;
+	if ( termId ) {
+		return termId;
+	}
+
+	return (
+		termNameToIdMap.get( searchTerm as string ) ||
```

Comment:
After 6a10f3fb5359ab455266c6c273d4cbd9c804f574, we don't need this line anymore, no? Because `termNameToIdMap` already includes lower-case terms.

Reply:
I think we need it because of [this reason](https://github.com/WordPress/gutenberg/blob/a993bfd8e108561bc2ca37a13415847d87c2019f/packages/block-library/src/query/edit/inspector-controls/taxonomy-controls.js#L36-L39). no? 🧐

Reply:
Oooh, right. Ignore my comment. :sweat: